src/coreclr/vm/debugdebugger.cpp

```diff
@@ -1254,4 +1607,192 @@ void DebugStackTrace::Element::InitPass2()
+    {
+        // Epilog entries are always just before the NO_MAPPING regions
+        return 1;
+    }
+    else if (ilOffsetB == (uint32_t)ICorDebugInfo::EPILOG)
+    {
+        // Epilog entries are always just before the NO_MAPPING regions
+        return -1;
+    }
+    else if (ilOffsetA < ilOffsetB)
+    {
+        return -1;
+    }
+    else
+    {
+        return 1;
+    }
+}
+bool ILToNativeMapArrays::CompareLessOffsets(uint32_t ilOffsetA, uint32_t nativeOffsetA, uint32_t ilOffsetB, uint32_t nativeOffsetB)
+{
+    int32_t ilOffsetCompare = CompareILOffsets(ilOffsetA, ilOffsetB);
+    _ASSERTE(CompareILOffsets(ilOffsetB, ilOffsetA) == -ilOffsetCompare);
+
+    if (ilOffsetCompare != 0)
+    {
+        // If IL offsets are not equal, then we can return the result of the IL offset comparison.
+        return ilOffsetCompare < 0;
+    }
+    return nativeOffsetA < nativeOffsetB;
+}
+
+bool ILToNativeMapArrays::CompareGreaterOffsets(uint32_t ilOffsetA, uint32_t nativeOffsetA, uint32_t ilOffsetB, uint32_t nativeOffsetB)
+{
+    int32_t ilOffsetCompare = CompareILOffsets(ilOffsetA, ilOffsetB);
+    _ASSERTE(CompareILOffsets(ilOffsetB, ilOffsetA) == -ilOffsetCompare);
+
+    if (ilOffsetCompare != 0)
+    {
+        // If IL offsets are not equal, then we can return the result of the IL offset comparison.
+        return ilOffsetCompare > 0;
+    }
+    return nativeOffsetA > nativeOffsetB;
+}
+
+void ILToNativeMapArrays::Swap(uint32_t* rguiNativeOffset, uint32_t *rguiILOffset, int32_t i, int32_t j)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    uint32_t tempNative = rguiNativeOffset[i];
+    uint32_t tempIL = rguiILOffset[i];
+    rguiNativeOffset[i] = rguiNativeOffset[j];
+    rguiILOffset[i] = rguiILOffset[j];
+    rguiNativeOffset[j] = tempNative;
+    rguiILOffset[j] = tempIL;
+}
+
+void ILToNativeMapArrays::Copy(uint32_t* rguiNativeOffset, uint32_t *rguiILOffset, int32_t i, int32_t j)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    rguiNativeOffset[i] = rguiNativeOffset[j];
+    rguiILOffset[i] = rguiILOffset[j];
+}
+
+int32_t ILToNativeMapArrays::Partition(uint32_t* rguiNativeOffset, uint32_t *rguiILOffset, int32_t low, int32_t high)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    // Choose the pivot as the middle element and 
+    // pull the pivot the the end of the array.
+    Swap(rguiNativeOffset, rguiILOffset, low + (high - low) / 2, high);
+
+    uint32_t pivotNative = rguiNativeOffset[high];
+    uint32_t pivotIL = rguiILOffset[high];
+    int i = low - 1;
+
+    for (int j = low; j < high; j++)
+    {
+        if (CompareLessOffsets(rguiILOffset[j], rguiNativeOffset[j], pivotIL, pivotNative))
+        {
+            i++;
+            Swap(rguiNativeOffset, rguiILOffset, j, i);
+        }
+    }
+
+    // Place pivot in the correct position
+    Swap(rguiNativeOffset, rguiILOffset, i+1, high);
+    return i + 1;
+}
+
+void ILToNativeMapArrays::QuickSort(uint32_t* rguiNativeOffset, uint32_t *rguiILOffset, int32_t low, int32_t high, int32_t stackDepth)
```

Comment:
I see that utilcode has a CQuickSort implementation. Should we be leveraging that or there is some reason it was unsuitable?